File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string CurrentUserName { get; set; }
        public string Gender { get; set; }
        public string PrintStatus { get; set; }
        public string SearchUser { get; set; }
        public int SearchUserID { get; set; }
        public string SearchPrinter { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 150;
        public string OrderBy { get; set; } = "lastActive";
    }
}

File: client/src/app/_models/userParams.ts
import { User } from './user';

export class UserParams {
    gender: string;
    printStatus: string;
    searchUser: string;
    searchPrinter: string;
    minAge = 18;
    maxAge = 99;
    pageNumber = 1;
    pageSize = 12;
    printJobPageNumber = 1;
    printJobPageSize = 10;
    printerPageNumber = 1;
    printerPageSize = 5;
    accessPageNumber = 1;
    accessPageSize = 10;
    semesterPageNumber = 1;
    semesterPageSize = 10;
    classlistPageNumber = 1;
    classlistPageSize = 10;
    urlRequestsPageSize = 5;
    orderBy = 'lastActive';
    searchYear: string;
    searchTerm: string;
    semesterId: number;
    

    constructor(user: User) {
        this.gender = user.gender === 'female' ? 'male' : 'female';
        this.printStatus = 'Held';
        this.searchUser = "";
        this.searchPrinter = "";
        this.searchYear = "All";
        this.searchTerm = "All";
    }
}

